utils/api_auth.py

```diff
@@ -36,5 +37,6 @@ def is_api_key_valid(api_key) -> bool:
     """
     allowed_api_keys = os.getenv("API_AUTH_ALLOWED_KEYS").split(" ")
     return any(
-        check_hash(api_key, allowed_api_key) for allowed_api_key in allowed_api_keys
+        check_hash(api_key, bytes.fromhex(allowed_api_key))
+        for allowed_api_key in allowed_api_keys
```

Comment:
@Opeyem1a
We are only calling the apis from our backend, right?

Reply:
@keyvankhademi 

Currently the list of consumers is:
- Our backend
- Our simple API frontend (something for demos)

We intend this API to be exgtensible for consumption at other institutions (i.e. Harvard Business School discussed wanting to integrate with our algorithms and have already developed an internal tool for this that could connect to our API instead of whatever they are currently doing)